### ISSUE TITLE
Running polymer demos in current Firefox (61.0) fails due to incorrect version of webcomponentsjs.jar (1.2 is OK, 2.0 fails)

### ISSUE THREAD

**Issue by kkolinko** [vaadin/flow-demo#453]
Demos using polymer elements, e.g. `demo-minesweeper-polymer`, display a blank screen when I try to run them in current Firefox (61.0). I tracked down that this is solved by replacing `webcomponentsjs.jar` with a correct version.

Configuration:
* `master` branch, also tried tag `1.0.0` - no difference
* Apache Maven 3.5.4

Steps to reproduce:
1. Go into `demo-minesweeper-polymer` module.
2. Run it with jetty, e.g. `mvn jetty:run`
3. Launch Firefox and open http://localhost:8080/
4. Expected: Minesweper game.
Actual: blank screen.

The war built for this web application includes the following library: `webcomponentsjs-2.0.0-beta.3.jar`. The problem persists if I try any `2.0.x` version of the library: 2.0.0, 2.0.2. (2.0.2 is currently broken - I reported the issue to webjars project, see [[1]](https://github.com/webjars/webjars/issues/1786), I repacked the jar file manually). The problem goes away if I use `webcomponentsjs-1.2.2.jar`.

Looking at network tab in developer tools in Firefox, with version 2.0.2 the first requests are the following:
```
http://localhost:8080/
http://localhost:8080/frontend/bower_components/webcomponentsjs/webcomponents-loader.js
http://localhost:8080/VAADIN/static/client/client-BF4676DDAFDA5E40AE93EFEB692AAAD5.cache.js
http://localhost:8080/frontend/bower_components/webcomponentsjs/bundles/webcomponents-sd-ce.js
```
With version 1.2.2 the 4th request is
```
http://localhost:8080/frontend/bower_components/webcomponentsjs/webcomponents-hi-sd-ce.js
```
The difference in name, "hi", means that a polyfill for HTML Imports is included.

Running `mvn dependency:tree`, the relevant fragment from the report is the following:
```
[INFO] +- org.webjars.bowergithub.polymer:polymer:jar:2.5.0:compile
[INFO] |  +- org.webjars.bowergithub.webcomponents:shadycss:jar:1.3.1:compile (version selected from constraint [1.1.0,2))
[INFO] |  \- org.webjars.bowergithub.webcomponents:webcomponentsjs:jar:2.0.0-beta.3:compile (version selected from constraint [1.0.19,2))
```

So Maven accepted version "2.0.0-beta.3" as valid version in the range "[1.0.19,2))".

Updating `org.webjars.bowergithub.polymer:polymer:jar` from the above version `2.5.0` to current `2.6.0` results in the following report:
```
[INFO] +- org.webjars.bowergithub.polymer:polymer:jar:2.6.0:compile
[INFO] |  +- org.webjars.bowergithub.webcomponents:shadycss:jar:1.3.1:compile (version selected from constraint [1.1.0,2))
[INFO] |  \- org.webjars.bowergithub.webcomponents:webcomponentsjs:jar:2.0.0-beta.3:compile (version selected from constraint [1.1.0,2))
```

**Comment by kkolinko** [comment]
Adding the following element into `dependencyManagement/dependencies` list of the root `pom.xml` resolves this issue:

```
            <dependency>
                <groupId>org.webjars.bowergithub.webcomponents</groupId>
                <artifactId>webcomponentsjs</artifactId>
                <version>[1.1.0,1.9.9]</version>
            </dependency>
```
but I wonder whether there is a better way.

The current version sorting algorithm is documented in [[1]](https://maven.apache.org/ref/3.5.4/maven-artifact/apidocs/org/apache/maven/artifact/versioning/ComparableVersion.html) and [[2]](https://cwiki.apache.org/confluence/display/MAVENOLD/Versioning).

(Old versions of Maven used different sorting and were more strict in supported version numbers. An old document: [[3]](https://cwiki.apache.org/confluence/display/MAVENOLD/Dependency+Mediation+and+Conflict+Resolution#DependencyMediationandConflictResolution-DependencyVersionRanges),  An old document with an example of broken sorting due to too many numbers: [[4]](https://docs.oracle.com/middleware/1212/core/MAVEN/maven_version.htm#MAVEN400))

**Comment by SomeoneToIgnore** [comment]
Good catch, thanks.
We use the fixed `1.2.0` version in the platform bom, so it's just the internal demos issue.
PR: https://github.com/vaadin/flow-demo/pull/454
